Sources/Workspace/Workspace.swift

```diff
@@ -779,7 +779,7 @@ extension Workspace {
         let lock = Lock()
         let sync = DispatchGroup()
         var rootManifests = [Manifest]()
-        packages.forEach { package in
+        Set(packages).forEach { package in
```

Comment:
small optimization

Reply:
It is a bit unexpected to have duplicate entries here, perhaps they should be diagnosed earlier if that's the case?

Reply:
There's no guarantee that libSwiftPM clients deduplicate the root packages, so this change probably makes sense.  The client I do know about does dedup them though, before calling libSwiftPM.